cmake/AutoDetectF16C.cmake

```diff
@@ -0,0 +1,45 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
```

Comment:
Add an include guard

Reply:
It seems the include_guard function is only available from cmake 3.12 - the minimum defined is 3.0.2

Shouldn't `if(NOT DEFINED SUPPORT_F16C)` be enough of an include guard anyway?

Reply:
Yes, unfortunately. But for now you can do:
```cmake
if(AUTO_DETECT_F16_CMAKE_INCLUDED)
  return()
endif()

set(AUTO_DETECT_F16_CMAKE_INCLUDED True)

```

Reply:
I've done the changes, but then `if(NOT DEFINED SUPPORT_F16C)` felt a little superfluous, so I removed. Wound't `if(NOT DEFINED SUPPORT_F16C)` have been enough of an include guard, though? Maybe we can talk about this tomorrow during the office hours...

Reply:
Sure) it's called [recurring Apache MXNet (Incubating) User Groups](https://cwiki.apache.org/confluence/display/MXNET/Apache+MXNet+%28Incubating%29+User+Groups+recurring+meetings) now though :)